diff -cNr ../orig/coreutils-8.19/lib/fseeko.c ./lib/fseeko.c
*** ../orig/coreutils-8.19/lib/fseeko.c	2012-07-04 20:13:02.000000000 +0900
--- ./lib/fseeko.c	2013-04-24 00:34:04.743346046 +0900
***************
*** 15,21 ****
     with this program; if not, see <http://www.gnu.org/licenses/>.  */
  
  #include <config.h>
! 
  /* Specification.  */
  #include <stdio.h>
  
--- 15,21 ----
     with this program; if not, see <http://www.gnu.org/licenses/>.  */
  
  #include <config.h>
! #include "file64.h"
  /* Specification.  */
  #include <stdio.h>
  
